feature/notifications

weichunnn opened the pull request.

NitinMa2 reviewed (changes requested):
@weichunnn  I had a look at the feature. It looks really good and I like how you integrated the classes and variables well. I do have a few comments I'd like you to look into before our next scrum meeting:

1. Your modal functionality is interesting but it introduced a bug. When the user logs out, it doesn't close the modal.
2. The notifications modal doesn't seem to be mobile responsive. 
3. Please convert all the static styles (10px, 100%, etc.) from TopNav, Notifications view and notifications module into scss variables.
4. I remember a red badge showing in TopNav when there are unread notifications. It's not showing now. Potential bug?
5. In the Router, make sure to add the `meta.requiresAuth` check as done for similar pages.
6. Let's not use h1 & p tags. You used the classes correctly but as indicated in the [developer guides](https://eureka-guide.netlify.app/#typography), it needs to be a `div`.
7. Add the class `page-pad` to the notifications page
8. Is there any reason the DB structure hasn't been flattened with a new `type` or `category` attribute? I think this point has been clearly brought up by Blaise and myself quite a few times.
9. You specified the Notification Type under `AppTypes`. I think it is more relevant to put it under `FirebaseTypes` and make sure the structure matches that of the DB. Some refactoring needed here. For the Notification Types you can create something like how `UserRoles` was done unless you know of something better.
10. Please create a function that creates a new notification by writing it into the DB. This function should accept some fixed parameters like from, to, etc. and can be imported into store. This is a good way of abstracting the implementation details for developers who need to send a notification so they wouldn't have to write a new DB write command everytime.
11. You can incorporate your emailJS code with this branch and have functionality to send out the emails.

Do let me know if I can clarify anything further or if I can support you in any way regarding these comments.

weichunnn commented:
Hi @NitinMa2 , based on your feedback, I had made the necessary changes. Do take a look and let me know if anything is breaking. From my side, everything looks good to me. 

1. Refactored to fix.
2. Fixed. Modal  has a maximum size of 300 px on small screen, hence this wouldn’t be any problem on phone screens. 
3. Need more clarifications on this. 
4. This is working on my machine. Maybe you can point out a case where it’s breaking on yours. 
5. Done
6. Refactored to show the changed. However, we should consider usability and accessibility (to be more user focused than dev focused) [What is the difference between p and div?](shorturl.at/uCDF1)
7. Added page pad, however I override the value to reduce wasted spaces
8. DB structure had been flattened as per our discussion
9. Done
10. Done
11. Done

NitinMa2 reviewed (approved):
Great stuff on this feature @weichunnn ! Thanks for updating the relevant section of the documentation as well. I'm merging this branch.
